include/arch/arm/arch/machine/gic_v3.h

```diff
@@ -335,19 +335,23 @@ static inline void maskInterrupt(bool_t disable, irq_t irq)
 
 static inline void ackInterrupt(irq_t irq)
 {
-    word_t hw_irq = IRQT_TO_IRQ(irq);
-    assert(IS_IRQ_VALID(active_irq[CURRENT_CPU_INDEX()]) && (active_irq[CURRENT_CPU_INDEX()] & IRQ_MASK) == hw_irq);
+    assert(IS_IRQ_VALID(active_irq[CURRENT_CPU_INDEX()]) && (active_irq[CURRENT_CPU_INDEX()] & IRQ_MASK) == IRQT_TO_IRQ(irq));
 
-    if (is_irq_edge_triggered(hw_irq)) {
-        gic_pending_clr(hw_irq);
-    }
```

Comment:
I’ve never understood this part. Any reason why we were clearing the IRQ pending state?

This seems to be broken and leads to miss IRQs maybe I’m missing something.

Reply:
I think this is a mistake in the code due to using the masking approach.  The split EOI mode should avoid needing to do this.'